### DISCUSSION TITLE
占用内存比较大

### DISCUSSION THREAD

**Discussion by yugu91** [messense/aliyundrive-webdav#140] [Q&A] [answered]
### 问题描述

在软路由上2g物理内存和2g swap，上传40g的20个文件大概是3m的上传速度，通过Rclone挂载Webdav上传，内存占用了14%

### 版本

0.5.5

### 运行平台

centos8

**Comment by messense** [comment, marked as answer]
VIRT 还是 RES 占用？提供下截图？

**Reply by yugu91** [reply]
<img width="803" alt="截屏2021-11-10 上午11 17 05" src="https://user-images.githubusercontent.com/10072650/141043566-d7733b14-d491-417a-9e42-f5a34326f5b5.png">

大概是这个样子，平均是13% - 15%占用

**Reply by messense** [reply]
差不多，目前上传一个文件 buffer 是 10MB，20 * 10MB = 200 MB 的 buffer 占用，上面 RES ~= 279 MB

**Reply by messense** [reply]
可以优化，不过需要改造 https://github.com/miquels/webdav-handler-rs 适配，我没有上传需求所以优化上传优先级比较低。有强需求可以考虑 https://github.com/messense/aliyundrive-webdav/discussions/126
